irohad/ordering/impl/on_demand_ordering_service_impl.cpp

```diff
@@ -118,34 +118,26 @@ OnDemandOrderingServiceImpl::onRequestProposal(consensus::Round round) {
  * continues getting all the transactions from the ongoing batch until the
  * required amount is collected.
  * @param requested_tx_amount - amount of transactions to get
- * @param tx_batches_queue - the queue to get transactions from
+ * @param batch_collection - the collection to get transactions from
  * @param discarded_txs_amount - the amount of discarded txs
 
  * @return transactions
  */
 static std::vector<std::shared_ptr<shared_model::interface::Transaction>>
 getTransactions(size_t requested_tx_amount,
-                tbb::concurrent_queue<TransactionBatchType> &tx_batches_queue,
+                detail::BatchSetType &batch_collection,
                 boost::optional<size_t &> discarded_txs_amount) {
-  TransactionBatchType batch;
   std::vector<std::shared_ptr<shared_model::interface::Transaction>> collection;
-  std::unordered_set<std::string> inserted;
 
-  while (collection.size() < requested_tx_amount
-         and tx_batches_queue.try_pop(batch)
-         and inserted.insert(batch->reducedHash().hex()).second) {
+  for (auto it = batch_collection.begin();
+       it != batch_collection.end() and collection.size() < requested_tx_amount;
+       ++it) {
     collection.insert(
         std::end(collection),
-        std::make_move_iterator(std::begin(batch->transactions())),
-        std::make_move_iterator(std::end(batch->transactions())));
-  }
-
-  if (discarded_txs_amount) {
-    *discarded_txs_amount = 0;
-    while (tx_batches_queue.try_pop(batch)) {
-      *discarded_txs_amount += boost::size(batch->transactions());
```

Comment:
why did you remove this? now the `discarded_txs_amount` it not set.

Reply:
It was expected to be replaced with atomic counter in `onBatches` method, so that there are no transactions to discard. However, since it is not introduced yet, I would suggest to make the following changes:
```C++
  auto it = tx_batches_queue.begin();
  for (;
       it != tx_batches_queue.end() and collection.size() < requested_tx_amount;
       ++it) {
    collection.insert(std::end(collection),
                      std::begin((*it)->transactions()),
                      std::end((*it)->transactions()));
  }

  if (discarded_txs_amount) {
    *discarded_txs_amount = 0;
    for (; it != tx_batches_queue.end(); ++it) {
      *discarded_txs_amount += boost::size((*it)->transactions());
    }
  }
```

Reply:
whoops. please either remove the discarded txs counter at all, or set it in `getTransactions`